app/helpers/repositories_helper.rb

```diff
@@ -44,6 +44,17 @@ def access_options
   end
 
   def repository_modal_body
-    modal_body(t("delete_repository"), t("delete_repository_desc"), (controller_name == "repositories" ? resource : parent), t("repository.delete"))
+    modal_body(t('delete_repository'), t('delete_repository_desc'),
+      reource_repository, t('repository.delete'))
+  end
+
+  def repository_undelete_modal_body
+    modal_body(t('repository.undelete.headline'),
+      t('repository.undelete.description'), [reource_repository, :undestroy],
+      t('repository.undelete.button'), method: :post, btn_class: 'btn-primary')
+  end
+
+  def reource_repository
+    controller_name == 'repositories' ? resource : parent
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.